src/pokeblock.c

```diff
@@ -0,0 +1,1105 @@
+            gMain.state++;
+            break;
+        case 14:
+            sub_810BC98();
+            sub_810BD08();
+            gMain.state++;
+            break;
+        case 15:
+            sub_810BB0C();
+            sub_810BB30();
+            sub_810BC84(gUnknown_02039248[1]);
+            gMain.state++;
+            break;
+        case 16:
+            ime = REG_IME;
+            REG_IME = 0;
+            REG_IE |= INTR_FLAG_VBLANK;
+            REG_IME = ime;
+            REG_DISPSTAT |= DISPSTAT_VBLANK_INTR;
+            SetVBlankCallback(sub_810B68C);
+            REG_DISPCNT = DISPCNT_OBJ_ON | DISPCNT_BG2_ON | DISPCNT_BG1_ON | DISPCNT_BG0_ON | DISPCNT_OBJ_1D_MAP;
+            gMain.state++;
+            break;
+        case 17:
+            if (sub_8055870() != TRUE)
+            {
+                gMain.state++;
+            }
+            break;
+        case 18:
+            BeginNormalPaletteFade(-1, 0, 0x10, 0, 0);
+            gPaletteFade.bufferTransferDisabled = FALSE;
+            SetMainCallback2(sub_810B674);
+            return TRUE;
+    }
+    return FALSE;
+}
+
+void sub_810B96C(void)
+{
+    do {
+        if (sub_810B6C0() == TRUE)
+        {
+            CreateTask(sub_810BF7C, 0);
+            break;
+        }
+    } while (sub_80F9344() != TRUE);
+}
+
+static bool8 sub_810B998(void)
+{
+    switch (ewram[0x1ffff])
+    {
+        case 0:
+            LZDecompressVram(gMenuPokeblock_Gfx, (u8 *)BG_CHAR_ADDR(2));
+            ewram[0x1ffff]++;
+            break;
+        case 1:
+            sub_800D238(gMenuPokeblock_Tilemap, gBGTilemapBuffers[2]);
+            ewram[0x1ffff]++;
+            break;
+        case 2:
+            LoadCompressedPalette(gMenuPokeblock_Pal, 0, 0xc0);
+            ewram[0x1ffff]++;
+            break;
+        case 3:
+            LoadCompressedObjectPic(&gUnknown_083F7F74);
+            ewram[0x1ffff]++;
+            break;
+        case 4:
+            LoadCompressedObjectPalette(&gUnknown_083F7F7C);
+            ewram[0x1ffff] = 0;
+            return TRUE;
+    }
+    return FALSE;
+}
+
+u8 sub_810BA50(s16 x, s16 y, u8 subpriority)
+{
+    return CreateSprite(&gSpriteTemplate_83F7F84, x, y, subpriority);
+}
+
+void sub_810BA7C(u8 a0)
+{
+    gUnknown_02039244 = a0;
+    switch (gUnknown_02039244)
+    {
+        default:
+            gUnknown_03000758 = gUnknown_083F7F1C;
+            gUnknown_0203924C = sizeof gUnknown_083F7F1C;
+            break;
+        case 2:
+            gUnknown_03000758 = gUnknown_083F7F1F;
+            gUnknown_0203924C = sizeof gUnknown_083F7F1F;
+            break;
+        case 3:
+            gUnknown_03000758 = gUnknown_083F7F21;
+            gUnknown_0203924C = sizeof gUnknown_083F7F21;
+            break;
+    }
+}
+
+void sub_810BADC(void)
+{
+    sub_810BA7C(2);
+    SetMainCallback2(sub_810B96C);
+}
+
+void sub_810BAF4(void)
+{
+    sub_810BA7C(3);
+    SetMainCallback2(sub_810B96C);
+}
+
+static void sub_810BB0C(void)
```

Comment:
In case you want to tackle it, there's a small `#ifdef DEBUG` function just before `sub_810BB0C`.

```
	thumb_func_start debug_sub_8120F98
debug_sub_8120F98:
	push	{r4, r5, lr}
	mov	r3, #0x0
	ldr	r1, _132
	ldrb	r0, [r1]
	cmp	r0, #0
	beq	_130	@cond_branch
	ldr	r5, _132 + 4
	add	r4, r1, #0
_131:
	lsl	r0, r3, #0x3
	add	r2, r0, r5
	add	r0, r0, r4
	ldr	r1, [r0, #0x4]
	ldr	r0, [r0]
	str	r0, [r2]
	str	r1, [r2, #0x4]
	add	r0, r3, #1
	lsl	r0, r0, #0x18
	lsr	r3, r0, #0x18
	cmp	r3, #0x27
	bhi	_130	@cond_branch
	lsl	r0, r3, #0x3
	add	r0, r0, r4
	ldrb	r0, [r0]
	cmp	r0, #0
	bne	_131	@cond_branch
_130:
	pop	{r4, r5}
	pop	{r0}
	bx	r0
_133:
	.align	2, 0
_132:
	.word	gUnknown_083F7F9C
	.word	gSaveBlock1 + 0x7f8
	thumb_func_end debug_sub_8120F98
```